### PR TITLE
Implement asynchronous support ODataJsonLightCollectionDeserializer

### PR DESCRIPTION
**Pull request by gathogojr**
### Issues

This pull request partially fulfills #2019.

### Description

Implement asynchronous support in **`ODataJsonLightCollectionDeserializer`**.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Comment by gathogojr** [comment]
Marked PR as draft until related PR is merged and rebased

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs [hunk 1]
```diff
@@ -278,15 +281,289 @@ internal void ReadCollectionEnd(bool isReadingNestedPayload)
...
+        internal async Task<Tuple<ODataCollectionStart, IEdmTypeReference>> ReadCollectionStartAsync(
+            PropertyAndAnnotationCollector collectionStartPropertyAndAnnotationCollector,
+            bool isReadingNestedPayload,
+            IEdmTypeReference expectedItemTypeReference)
+        {
+            this.JsonReader.AssertNotBuffering();
+
+            IEdmTypeReference actualItemTypeReference = expectedItemTypeReference;
+
+            ODataCollectionStart collectionStart = null;
```
Comment (KenitoInc) [inline src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs:314]:
```suggestion
            ODataCollectionStart collectionStart = null;

```

**Review by KenitoInc — APPROVED** [review]
one minor fix.
Otherwise LGTM

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs (hunk 1 above)
Comment (gathogojr) [reply src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs:314]: Fixed

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs [hunk 2]
```diff
@@ -72,6 +73,7 @@ internal ODataJsonLightCollectionDeserializer(ODataJsonLightInputContext jsonLig
                 while (this.JsonReader.NodeType == JsonNodeType.Property)
                 {
                     IEdmTypeReference actualItemTypeRef = expectedItemTypeReference;
+                    this.ReadPropertyCustomAnnotationValue = this.ReadCustomInstanceAnnotationValue;
```
Comment (ElizabethOkerio) [inline src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs:76]: Haven't understood this assignment. if they are the same why can't we use <code>this.ReadCustonInsanceAnnotationValue</code> in the place of <code>this,ReadPropertyCustomAnnotationValue </code>

**Review by ElizabethOkerio — APPROVED** [review]

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs (hunk 2 above)
Comment (gathogojr) [reply src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs:76]: @ElizabethOkerio `ReadPropertyCustomAnnotationValue` is a `Func` property declared in the base class `ODataJsonLightDeserializer`. The derived classes are required to initialize it with a function that does the actual reading of custom property annotation value. That is what we're doing here.

Comment (ElizabethOkerio) [reply src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs:76]: I think i didn't look at their definitions

Comment (gathogojr) [reply src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionDeserializer.cs:76]: No problem. Happy to clarify

**Pull request merged**